pyrules/rules.py

```diff
@@ -1,93 +1,161 @@
-        if translations:
-            translator = Translator(translations)
-            for rule in self.rules:
-                for key in rule.keys():
-                    rule[key] = [translator.replace(item) for item in rule[key]]
-    def should_trigger(self, context):
-        return True
+        if name:
+            self.name = name
+        self._current_ruleid = None
+
     def perform(self, context):
         count = 0
         for rule in self.rules:
-            if all([eval(condition, context.as_dict) for condition in rule['if']]):
+            evaluator = LogicEvaluator(
+                rule['if'].get('logic'), rule['if']['conditions'])
+            if evaluator.evaluate(context):
                 count = count + 1
-                self._current_ruleid = rule.get('id', count)
+                self._current_ruleid = rule.get('rule') or count
                 for action, target in zip(rule['then'], rule['target']):
-                    if context._translations:
-                        action = context._translations.replace(action)
-                        target = context._translations.replace(target)
-                    result = context[target.replace('context.', '').strip()] = eval(action, context.as_dict)
+                    #if context._translations:
+                    #    action = context._translations.replace(action)
+                    #    target = context._translations.replace(target)
+                    result = \
+                        context[target.replace('context.', '').strip()] = (
+                            eval(action, context.as_dict)
+                            if isinstance(action, basestring)
+                            else action)
                     self.record(context, result)
             else:
-                break
+                continue
         else:
             self._current_ruleid = None
             return True
         return False
+
     @property
     def ruleid(self):
         if self._current_ruleid:
-            return "%s.%s" % (super(TableRuleset, self).ruleid, self._current_ruleid)
-        return super(TableRuleset, self).ruleid
-    
+            return "%s.%s" % (super(TableRule, self).ruleid, self._current_ruleid)
+        return super(TableRule, self).ruleid
+
+    @classmethod
+    def from_yaml(cls, text):
+        return cls._from_data(yaml.load(text))
+        
+    @classmethod
+    def from_json(cls, text):
+        return cls._from_data(json.loads(text))
+
+    @classmethod
+    def _from_data(cls, data):
+        # We have to convert non-string data in clauses back to strings,
+        # because they will be eval-ed
+        rules = []
+        for rule in data['rules']:
+            obj = {
+                'rule': rule.get('rule'),
+                'then': rule['then'],
+                'target': rule['target']}
+            if_clause = {}
+            # Convert conditions to dictionaries, i.e. "foo" becomes {"foo": True}
```

Comment:
I see your reservations.

The use of YAML is to make rule specs as much human readable/writable, as machine readable. So it is, in effect a DSL => same argument as for NaturalLanguageRule. So Table Rules are not generally meant to be output by a tool, but written by humans. Of course we can always add a nicer editor on top of it. 

For us English speaking developers writing variable names is straight forward an the simple approach. Not so much for e.g. German speaking business folks. We can argue about whether business people will and should ever specify rules, whatsoever, and in fact I doubt that, but they will want to read them, and it so so much simpler to show them a sentence like "if the journey is not on a weekend" => they get it and are able to say yes. OTH if you show them "if journey_daytype != weekend", or as it is now "if journey_daytype__ne : weekend". They just roll their eyes and call you a tech geek ;-)

So really having translations is a means to an end. So is catching errors in parsing the rules instead of at runtime.